test/e2e_node/disk_eviction_test.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	// The interval seconds between pod status checks
+	POD_CHECK_INTERVAL = 3
+
+	DUMMY_FILE = "dummy.file"
+)
+
+// TODO: Leverage config map to change kubelet eviction option based on available disk space in e2e tests.
+// To manually trigger the test for a node with disk space just over 15G :
+//   make test-e2e-node FOCUS="hard eviction test" TEST_ARGS="--eviction-hard=nodefs.available<15Gi"
+var _ = framework.KubeDescribe("Kubelet Eviction Manager [FLAKY] [Serial] [Disruptive]", func() {
+	f := framework.NewDefaultFramework("kubelet-eviction-manager")
+	var podClient *framework.PodClient
+	var c *client.Client
+	var n *api.Node
+
+	BeforeEach(func() {
+		podClient = f.PodClient()
+		c = f.Client
+		nodeList := framework.GetReadySchedulableNodesOrDie(c)
+		n = &nodeList.Items[0]
+	})
+
+	Describe("hard eviction test", func() {
+		Context("pod gets evicted when the disk usage is above the hard threshold", func() {
```

Comment:
nit : 
s/pod gets evicted when the disk usage is above the hard threshold/when container's disk usage exceeds the hard threshold